docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,287 @@
+# Proposal B - Extend Platform Variants and Introduce Compatibility Artifact
+
+This proposal recommends changes to image spec for more granular image selection and introduces a new image compatibility artifact type.
+Based on the artifact, cluster administrators can verify hosts or prepare the appropriate infrastructure for containers.
+
+## Modifications
+
+### Image Spec
+
+Extend the platform variants with new ARM and AMD64 micro-architecture levels.
+
+| ISA/ABI          | architecture | variant |
+|------------------|--------------|---------|
+| ...              | ...          | ...     |
+| ARM 64-bit, v9   | arm          | v9      |
+| AMD64 64-bit, v2 | amd64        | v2      |
+| AMD64 64-bit, v3 | amd64        | v3      |
+| AMD64 64-bit, v4 | amd64        | v4      |
+
+## Image Compatibility Artifact
+
+!["Image Compatibility Artifact"](img/B_artifact.png "Image Compatibility Artifact")
+
+### Definitions
+
+- image-compatibility artifact - an OCI artifact that stores the compatibility spec.
+- Compatibility Spec - a json object that represents a graph of namespaces.
+- Namespace - a validation area under which users can describe plugin specs that are consumed by plugins (executable binaries) as input.
+- Plugin - an executable binary that extracts data and checks compatibility on a host.
+
+A new imageCompatibility artifact type that contains a layer with a JSON compatibility object.
+The object represents a graph of namespaces.
+Namespace describes the validation area and is another object containing data that provides plugins specification for the corresponding plugins.
+A plugin is an executable binary that validates the host and returns information about success or failure.
```

Comment:
> Are plugins run in a sandbox?

For Linux, I planned to use SELinux, AppArmor or Firejail 

>  Would that have root access to the host?

The plugins developed by us would not have root access, we would have to create appropriate sandbox for them.
However, I cannot speak for the community how they use and develop plugins.

> Would this be useful as an attack vector by malicious parties?

It could be. If we don't verify community plugins.

My initial idea was to allow for out-of-tree plugins and give a freedom to users to develop and run what they want.
However, that comes with many security risks, problems of binaries validation and verification. It gets too complex for this project. I am going to propose something less open.

> And if this is being used for node provisioning, then a binary to verify the node would be the wrong stage of the workflow.

It's not for node provisioning.